roles/default/kiali-deploy/tasks/main.yml

```diff
@@ -285,11 +285,22 @@
   when:
     kiali_vars.external_services.tracing.in_cluster_url == "" and is_maistra
 
-- name: Set default Tracing in_cluster_url
+- name: Set default Tracing in_cluster_url for grpc consumption
+  set_fact:
+    kiali_vars: "{{ kiali_vars | combine({'external_services': {'tracing': {'in_cluster_url': 'http://tracing.' + kiali_vars.istio_namespace + ':16685/jaeger'}}}, recursive=True) }}"
+  when:
+  - is_maistra == False
+  - kiali_vars.external_services.tracing.in_cluster_url == ""
+  - kiali_vars.external_services.tracing.use_grpc is not defined or kiali_vars.external_services.tracing.use_grpc == True
+
+- name: Set default Tracing in_cluster_url for http consumption
   set_fact:
     kiali_vars: "{{ kiali_vars | combine({'external_services': {'tracing': {'in_cluster_url': 'http://tracing.' + kiali_vars.istio_namespace + '/jaeger'}}}, recursive=True) }}"
   when:
-    kiali_vars.external_services.tracing.in_cluster_url == "" and is_maistra == False
+  - is_maistra == False
```

Comment:
I just noticed this - why do we have is_maistra here? If we are running in Maistra but we are not created by the Maistra operator, don't we want this default?

If created by the maistra operator, this is already set, and the `kiali_vars.external_services.tracing.in_cluster_url == ""` will be false and this still won't be executed.

I have a feeling we need to take out these is_maistra==False conditionals because it will most likely screw up devs when they deploy their own Kiali CR in a maistra environment

Reply:
To be completely honest, I don't know. Just expanded the existing expression :sweat_smile: 
It sounded that we didn't want to do the port differentiation in maistra because it uses different ports.
If the tasks are executed in order of declaration, this maistra condition wouldn't be necessary however it makes it fragile.
The role `Set default Tracing in_cluster_url for Maistra` sets the url for maistra. The other Tracing roles do it for upstream (and specifically have the condition of !is_maistra)

Reply:
> The role `Set default Tracing in_cluster_url for Maistra` sets the url for maistra.

Oh, yeah, I didn't notice that - the existing task above will set this "if Maistra" so the following tasks won't be invoked if on Maistra anyway since that variable will have been set already and will not be null.

OK. We can leave this.